packages/core/app/mirage/routes/report.js

```diff
@@ -8,17 +8,39 @@ export default function() {
    * reports/ - GET endpoint to fetch many reports
    */
   this.get('/reports', function({ reports }, request) {
+    let reportsObject = null;
     let idFilter = request.queryParams['filter[reports.id]'];
+    let queryFilter = request.queryParams['filter[reports]'];
 
     // Allow filtering
     if (idFilter) {
       let ids = idFilter.split(',');
-      reports = reports.find(ids);
+      reportsObject = reports.find(ids);
+    } else if (queryFilter) {
+      let filterParameters = queryFilter
+        .split(';')[0]
+        .replace(/\(/g, '')
+        .replace(/\)/g, '')
+        .replace(/\*/g, '')
+        .split(',')
+        .map(el => el.split('==')[1]);
+      let author = queryFilter.includes('author')
+        ? queryFilter
```

Comment:
maybe a similar function could be invoked here? `getQueryAuthor`. It could return `undefined` or `null` whichever we think is best. Then there wouldn't be a need for the turnery here